week2/homework/src/index.js

```diff
@@ -1,3 +1,46 @@
 'use strict';
+const {
+    add,
+    reset,
+    list,
+    remove,
+    update
+} = require("./IOutil");
 
-// TODO: Write the homework code in this file
+const TODO_FILE = 'todo.json';
+
+function main() {
+    const [, , cmd, ...args] = process.argv;
+
+    switch (cmd) {
+        case 'add':
+            add(TODO_FILE, "utf8", args);
```

Comment:
You don't accept the encoding as an argument in your other functions (`list`, `reset`, etc...). This is a bit confusing. Both are valid, just choose one single approach for consistency.

Reply:
Agreed, the encode should be an argument for other functions too